modin/utils.py

```diff
@@ -490,46 +491,46 @@ def wrapped(*args: tuple, **kw: dict) -> Any:
     return decorator
 
 
-# TODO add proper type annotation
-def to_pandas(modin_obj: SupportsPrivateToPandas) -> Any:
+def func_from_deprecated_location(
+    func_name: str, module: str, deprecation_message: str
+) -> Callable:
     """
-    Convert a Modin DataFrame/Series to a pandas DataFrame/Series.
+    Create a function that decorates a function ``module.func_name`` with a ``FutureWarning``.
 
     Parameters
     ----------
-    modin_obj : modin.DataFrame, modin.Series
-        The Modin DataFrame/Series to convert.
+    func_name : str
+        Function name to decorate.
+    module : str
+        Module where the function is located.
+    deprecation_message : str
+        Message to print in a future warning.
 
     Returns
     -------
-    pandas.DataFrame or pandas.Series
-        Converted object with type depending on input.
+    callable
     """
-    return modin_obj._to_pandas()
 
+    def deprecated_func(*args: tuple[Any], **kwargs: dict[Any, Any]) -> Any:
```

Comment:
Should we use @functools.wraps(func) here or assign some attributes like __name__, __doc__ to deprecated_func from a deprecated function to be called?

Reply:
We can't do this. The code building a decorator is being executed at the time of the module's initialization, which means, that we cannot do circular imports at this point. See the simplified example:

```python
def create_deprecated(msg):
    # the code in this function is being executed at the time of module initialization,
    # meaning that the following line will trigger the circular import error:
    # from modin.pandas.io import to_pandas

    # can't use this as we haven't imported 'to_pandas' yet
    # @functols.wrap(to_pandas)
    def wrapper(*args, **kwargs):
        # we can only import actual 'to_pandas' here as this code is being executed
        # when the function is called and all the modules are initialized
        from modin.pandas.io import to_pandas
        return to_pandas(*args, **kwargs)

    return wrapper
   
to_pandas = create_deprecated("to_pandas")
```